packages/image_picker/image_picker_for_web/lib/image_resizer.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math';
+import 'dart:ui';
+import 'package:image_picker_for_web/image_resizer_utils.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+import 'dart:html' as html;
+
+/// Helper class that resizes image
+class ImageResizer {
+  /// Resizes the image if needed
+  /// Does not support gif image
+  Future<XFile> resizeImageIfNeeded(XFile file, double? maxWidth,
+      double? maxHeight, int? imageQuality) async {
+    if (!imageResizeNeeded(maxWidth, maxHeight, imageQuality) ||
+        file.mimeType == "image/gif") {
+      //TODO Implement maxWidth and maxHeight for image/gif
+      return file;
+    }
+    try {
+      final imageElement = await loadImage(file.path);
+      final canvas = resizeImageElement(imageElement, maxWidth, maxHeight);
+      return writeCanvasToFile(file, canvas, imageQuality);
+    } catch (e) {
+      return file;
```

Comment:
Note that you're revoking the blobUrl in the onLoad event of the `loadImage` method. This `return` is going to return a bad file in case `resizeImageElement` or `writeCanvasToFile` fail in any way.

I think you should `revokeObjectUrl` only after successfully calling `writeCanvasToFile` to ensure that the old blob URL is revoked after nobody else is going to use it.

Reply:
(This is unresolved, I explained the issue in another comment)